### ISSUE TITLE
`steps` argument in `Mat::new_nd_with_data()` has a wrong type

### ISSUE THREAD

**Issue by jerry73204** [twistedfall/opencv-rust#201]
The `Mat::new_nd_with_data` ([doc](https://docs.rs/opencv/0.48.0/opencv/core/struct.Mat.html#method.new_nd_with_data)) has a `steps` argument with type `&size_t`. I passed `&0`, the suggested default value, and it failed with the message.

```
Error: OpenCV(4.5.1) /build/opencv/src/opencv-4.5.1/modules/core/src/matrix.cpp:259: error: (-13:Image step is wrong) Step 7954887940465321323 for dimension 1 must be a multiple of esz1 4 in function 'setSize'
 (code: -13)
```

Look like it treats the reference as an integer. I guess the argument should have the type `size_t`,

**Comment by jerry73204** [comment]
Also `new_nd_vec_with_data` has this issue.

**Comment by twistedfall** [comment]
I think it expects a null pointer, not a pointer to null. Can you please try passing something like `prt::null_ptr().as_ref()` for `steps`? I’ll try to improve generation in this are in any case

**Comment by jerry73204** [comment]
It won't work. [pointer.as_ref()](https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref) returns `None` when it is null.

**Comment by twistedfall** [comment]
You’re right! Well, some kind of `mem::transmute` should do the trick, but yeah, it needs some fixing.

**Comment by jerry73204** [comment]
It is my workaround to pass null pointer as reference. Unfortunately, it passes the tests in in debug build but is killed by SIGILL in release build.

```rust
use platform_types::size_t;
let step: &size_t = mem::transmute(ptr::null() as *const size_t);
core::Mat::new_nd_with_data(&size, typ, tensor.data_ptr(), step)?
```

This is my implementation FYI.
https://github.com/jerry73204/rust-cv-convert/blob/46136573897146f1ea183772b4daeee5ffecab97/src/with_opencv_tch.rs#L347-L357

**Comment by twistedfall** [comment]
In my tests it literally generates the asm `UD2` instruction starting with opt-level = 1. I suppose it has to do with the fact that references are guaranteed to be non-null in rust, so such transmute is instant UB. So it looks like there is no workaround. I'll try to make the fix asap.

**Comment by jerry73204** [comment]
Interesting, TIL there is a x86 instruction causes undefined behavior.

**Comment by twistedfall** [comment]
Can you please check the `rel` branch and confirm it fixes the problem for you? The `steps` argument is now `Option<&[size_t]>` so passing `None` should work.

**Comment by jerry73204** [comment]
Great. It passes every test even in release build.

**Comment by twistedfall** [comment]
It's now released in v0.49.0
